tests/Doctrine/Tests/DBAL/Query/Expression/CompositeExpressionTest.php

```diff
@@ -77,4 +77,12 @@ public function provideDataForConvertToString()
             ),
         );
     }
+
+    public function testAddPreventsAddingEmptyCompositeExpressionObjects()
+    {
+        $emptyExpr = new CompositeExpression(CompositeExpression::TYPE_AND, []);
+        $expr = new CompositeExpression(CompositeExpression::TYPE_AND, [$emptyExpr]);
+
+        $this->assertEquals(0, count($expr));
```

Comment:
Ir that would be a way how the api can be used, why was there a check 

> ($part instanceof self && $part->count() > 0)

in the code? I think it shouldn't be possible to add constraints after adding the object to the CompositeExpression